service/tcp.go

```diff
@@ -370,12 +369,12 @@ func (h *tcpHandler) handleConnection(ctx context.Context, outerConn transport.S
 
 // Keep the connection open until we hit the authentication deadline to protect against probing attacks
 // `proxyMetrics` is a pointer because its value is being mutated by `clientConn`.
-func (h *tcpHandler) absorbProbe(clientConn io.ReadCloser, status string, proxyMetrics *metrics.ProxyMetrics) {
+func (h *tcpHandler) absorbProbe(clientConn transport.StreamConn, status string, proxyMetrics *metrics.ProxyMetrics) {
```

Comment:
Let's keep the APIs tight. Add an address field instead. A StreamConn is not needed, and may not have the right address. With PROXY protocol we will be getting the address from the client request instead.

Reply:
Humm, I take that back on the PROXY protocol. We probably don't want to use the entry point necessarily because it may vary from a large pool. That would blow up the metrics. Ideally we would have something like domain:port, but that's not possible.

Still, let's pass the address in this function.

Not necessarily for this PR, but you can consider creating that Client Connection object sooner. Then it can contain the actual connection, the client address (needed for PROXY), and other information like the listener ID or service ID.

Reply:
Done. Will create the client conn object in a separate PR though.